freqtrade/data/dataprovider.py

```diff
@@ -99,9 +101,10 @@ def get_analyzed_dataframe(self, pair: str, timeframe: str) -> Tuple[DataFrame,
             if self.runmode in (RunMode.DRY_RUN, RunMode.LIVE):
                 df, date = self.__cached_pairs[pair_key]
             else:
-                max_index = self.__slice_index
                 df, date = self.__cached_pairs[pair_key]
-                df = df.iloc[max(0, max_index - MAX_DATAFRAME_CANDLES):max_index]
+                if self.__slice_index is not None:
```

Comment:
Tbh i think that assert is probably better here. `self.__slice_index` should never be `None` in backtesting/hyperopt therefore it is a mistake in the bot. `if` hides this mistake and if this ever gets broken nobody will notice that and keep looking into last candle of entire range.

Reply:
I think youre going off of a wrong(ish) assumption.

You're right if you only look at the callbacks (`custom_stop`, `custom_sell`).

However, nothing prevents me from using `dp.get_analyzed_dataframe` in a `populate_sell` method (to get a different pair, for example).
If that'll work is a different topic (cached_pairs will not be assigned at that point yet) - but that's potentially small change ... so this will prepare for that scenario.

Reply:
Oh! `populate_sell` runs before main backtesting where slice index is set. I did not realize that 😅

Reply:
Well thats why it's called only once (per pair)... Which makes it fast, but also kinda limited in what you can do with it